contracts/modules/viewer/NexusViewer.sol

```diff
@@ -0,0 +1,117 @@
+  /// @param tokenIds An array of staking NFT token IDs associated with the member.
+  /// @return A ClaimableNxm struct containing details of the claimable NXM tokens.
+  function getClaimableNXM(address member, uint[] calldata tokenIds) external view returns (ClaimableNXM memory) {
+
+    // Governance
+    uint governanceRewards = _governance().getPendingReward(member);
+    
+    // Assessment
+    IAssessmentViewer.AssessmentRewards memory assessmentRewards = assessmentViewer.getRewards(member);
+    (uint assessmentStake, bool isStakeLocked) = _getAssessmentStake(member);
+
+    // Staking Pool
+    IStakingViewer.AggregatedTokens memory aggregatedTokens = stakingViewer.getAggregatedTokens(tokenIds);
+    uint managerTotalRewards = _getManagerTotalRewards(member);
+    
+    // V1
+    uint legacyPooledStakeDeposits = _legacyPooledStaking().stakerDeposit(member);
+    uint legacyPooledStakeRewards = _legacyPooledStaking().stakerReward(member);
+    uint legacyClaimAssessmentTokens = _tokenController().tokensLocked(member, "CLA");
+    (, , uint legacyCoverNoteDeposits) = _tokenController().getWithdrawableCoverNotes(member);
+    
+    return ClaimableNXM({
+      governanceRewards: governanceRewards,
+      assessmentRewards: assessmentRewards.withdrawableAmountInNXM,
+      assessmentStake: isStakeLocked ? 0 : assessmentStake,
+      stakingPoolTotalRewards: aggregatedTokens.totalRewards,
+      stakingPoolTotalExpiredStake: aggregatedTokens.totalExpiredStake,
+      managerTotalRewards: managerTotalRewards,
+      legacyPooledStakeRewards: legacyPooledStakeRewards,
+      legacyPooledStakeDeposits: legacyPooledStakeDeposits,
+      legacyCoverNoteDeposits: legacyCoverNoteDeposits,
+      legacyClaimAssessmentTokens: legacyClaimAssessmentTokens
+    });
+  }
+
+  /// @notice Retrieves the locked NXM tokens across the platform for a given member.
+  /// @dev Ensure the tokenIds passed belongs to the member address.
+  /// @param member The address of the member to query.
+  /// @param tokenIds An array of staking NFT token IDs associated with the member.
+  /// @return A StakedNXM struct containing details of the locked NXM tokens.
+  function getStakedNXM(address member, uint[] calldata tokenIds) external view returns (StakedNXM memory) {
+
+    IStakingViewer.AggregatedTokens memory aggregatedTokens = stakingViewer.getAggregatedTokens(tokenIds);
+
+    IAssessmentViewer.AssessmentRewards memory assessmentRewards = assessmentViewer.getRewards(member);
+    (uint assessmentStake, bool isStakeLocked) = _getAssessmentStake(member);
+
+    return StakedNXM({
+      stakingPoolTotalActiveStake: aggregatedTokens.totalActiveStake,
+      assessmentStake: isStakeLocked ? assessmentStake : 0,
+      assessmentRewards: assessmentRewards.totalPendingAmountInNXM - assessmentRewards.withdrawableAmountInNXM
+    });
+  }
+
+  function _getManagerTotalRewards(address member) internal view returns (uint managerTotalRewards) {
```

Comment:
Should you move this to StakingViewer? Might be useful more widely as a public function

Reply:
yes, its better that way. good suggestion 👍